### PR TITLE
Fix macOS bug where tray window causes spaces to switch

### LINKED ISSUE #2332: Client 3.0 on the Mac causes desktop/space switch
### Expected behaviour
When I click on the Nextcloud icon in the toolbar on a Mac with multiple desktops/spaces, the Nextcloud window should open on the current desktop.

### Actual behaviour
On a Mac with multiple desktops, when clicking the Nextcloud icon in the toolbar the mac will always switch to the desktop where the Nextcloud.app was launched.

I already checked the com.apple.spaces.plist to see if the app is assigned to a specific desktop and I couldn't find an assignment.

### Steps to reproduce
1. launch the Nextcloud 3.0 app on the first desktop
2. change to desktop/space 2
3. click on the nextcloud symbol in the menu bar

### Client configuration
Client version: 3.0.0

Operating system:
MacOS Catalina 10.15.6

OS language:
German

Installation path of client:
/Applications/

### PR DESCRIPTION
**Pull request by briankendall**
Closes #2332 

Let me know if there's any changes you need to the PR and I'll be happy to update my commit

### CODE REVIEW

File: src/gui/systray.h [hunk 1]
```diff
@@ -29,6 +29,7 @@ namespace OCC {
 #ifdef Q_OS_OSX
 bool canOsXSendUserNotification();
 void sendOsXUserNotification(const QString &title, const QString &message);
+void setTrayWindowLevelAndVisibleOnAllSpaces(QQuickWindow *qtWindow);
```
Comment (er-vin) [inline src/gui/systray.h:32]: I'd rename the parameter from qtWindow to window. Also seeing how it's used please change the type to QWindow.

File: src/gui/systray.mm [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #include <QString>
+#include <QQuickWindow>
```
Comment (er-vin) [inline src/gui/systray.mm:2]: Please include QWindow instead.

File: src/gui/systray.mm [hunk 3]
```diff
@@ -41,4 +42,13 @@ void sendOsXUserNotification(const QString &title, const QString &message)
     [notification release];
 }
 
+void setTrayWindowLevelAndVisibleOnAllSpaces(QQuickWindow *qtWindow)
```
Comment (er-vin) [inline src/gui/systray.mm:45]: Rename the parameter and change its type please.

File: src/gui/systray.mm [hunk 4]
```diff
@@ -41,4 +42,13 @@ void sendOsXUserNotification(const QString &title, const QString &message)
...
+{
+    NSView *view = (NSView *)qtWindow->winId();
+    NSPanel *window = (NSPanel *)[view window];
```
Comment (er-vin) [inline src/gui/systray.mm:48]: Please change window here to nativeWindow.

**Review by er-vin — CHANGES_REQUESTED** [review]

**Comment by briankendall** [comment]
Okay I made the requested changes. (And fixed my usage of `NSPanel` instead of `NSWindow` which wasn't technically correct.) Let me know if there's anything else you'd like me to do.

**Review by er-vin — APPROVED** [review]

**Pull request merged**